org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeBuilder.java

```diff
@@ -0,0 +1,71 @@
+package org.code.javabuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+
+/** The orchestrator for code compilation and execution. */
+public class CodeBuilder {
+  private final OutputAdapter outputAdapter;
+  private final InputAdapter inputAdapter;
+  private final ProjectFileManager fileManager;
+  private final File tempFolder;
+
+  public CodeBuilder(
+      InputAdapter inputAdapter, OutputAdapter outputAdapter, ProjectFileManager fileManager)
+      throws UserFacingException {
+    this.outputAdapter = outputAdapter;
+    this.inputAdapter = inputAdapter;
+    this.fileManager = fileManager;
+    try {
```

Comment:
This might be a little bit more work than we want to do in a constructor.

Reply:
Naively, this seems to make sense to have here. Do you have recommendations for setting it up differently?

Reply:
I think it's ok to leave it here for now and see how this code evolves.  Depending on how we split up compile and execute and how we pass the compiled class from one component to another, I could imagine a temp folder being initialized and cleaned up in function that calls both parts, for example.